### PR TITLE
Authorization with deezer

### PR DESCRIPTION
**Pull request by uaArsen**
Implement authorization with Deezer and refactor project structure 

1. Create controller, service, dto for authorization
2. Implement authorization via deezer api, create nessary configuration, utils and dto.
3. Add tests
4. Refactor soundcloud package -> dto/soundcloud
5. Refactor application.propertiese -> application.yml

### CODE REVIEW

File: server/src/main/java/org/room/apollo/server/service/AuthorizationService.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Service for authorization methods.
+ */
+@Service
+public class AuthorizationService {
+
+    private DeezerConfiguration deezerConfiguration;
```
Comment (AlexeyZatsepin) [inline server/src/main/java/org/room/apollo/server/service/AuthorizationService.java:22]: Can we autowire (inject) values of DeezerConfiguration and HttpUtil?

Comment (uaArsen) [reply server/src/main/java/org/room/apollo/server/service/AuthorizationService.java:22]: We are Autowiring through constructor.

File: server/src/main/java/org/room/apollo/server/util/HttpUtil.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import org.springframework.stereotype.Component;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+@Component
+public class HttpUtil {
```
Comment (AlexeyZatsepin) [inline server/src/main/java/org/room/apollo/server/util/HttpUtil.java]: Do we need to use HttpURLConnection? Is it better to remove this class and use ResTemplate?

Comment (AlexeyZatsepin) [reply server/src/main/java/org/room/apollo/server/util/HttpUtil.java]: RestTemplate

Comment (uaArsen) [reply server/src/main/java/org/room/apollo/server/util/HttpUtil.java]: Looks good, will change

**Review by AlexeyZatsepin — APPROVED** [review]

**Pull request merged**